src/mumble/PositionalAudioViewer.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_DIALOGS_POSITIONALAUDIOVIEWER_H_
+#define MUMBLE_MUMBLE_DIALOGS_POSITIONALAUDIOVIEWER_H_
+
+#include "PositionalData.h"
+
+#include <memory>
+
+#include <QDialog>
+#include <QTimer>
+
+namespace Ui {
+class PositionalAudioViewer;
+};
+
+class PositionalAudioViewer : public QDialog {
+public:
+	PositionalAudioViewer();
+	~PositionalAudioViewer();
+
+protected:
+	void update();
+	void updatePlayer(const PositionalData &data);
+	void updateCamera(const PositionalData &data);
+
+	QTimer m_timer;
+	std::unique_ptr< Ui::PositionalAudioViewer > m_ui;
```

Comment:
Something that just came to my mind: Usually Qt objects don't like if you `delete` them directly (which unique_ptr will do by default). Instead we should probably always prefer to call `deleteLater` on them and let Qt figure out the exact time of deletion.

That means that you'll have to use the unique_ptr with a custom deleter (use `deleteQObject` from `QtUtils.cpp`).

Reply:
Actually nvm. I just found out that the classes generated by Qt Designer aren't actually QObjects...

Reply:
The problem is that, as far as I know, Qt's memory manager is basically a garbage collector.

I would like the dialog object to be deleted instantly, instead.

We could use [Qt::WA_DeleteOnClose](https://doc.qt.io/qt-5/qt.html#WidgetAttribute-enum), but then we would have to set the pointer to `nullptr` manually.

Reply:
As I said for this case my argument doesn't apply. When dealing with actual QObjects though we don't have a choice. If you delete them directly, it can easily happen that the application Segfaults :shrug:

Reply:
I believe the class is a `QObject`, otherwise the signal-slot mechanism wouldn't work.

The reason why it doesn't cause issues is probably because we don't delete the objects ourselves, `Ui::PositionalAudioViewer`'s destructor does that.

Reply:
No these classes literally don't inherit `QObject`. You don't have to extend from `QObject` to use signals and slots. You only have to tell Qt's meta compiler that you would like to have this class QObject-like features using the `Q_OBJECT` macro ^^

Reply:
I see, thanks.